gobblin-data-management/src/main/java/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -190,6 +192,9 @@ private void publishFileSet(CopyEntity.DatasetAndPartition datasetAndPartition,
     long datasetOriginTimestamp = Long.MAX_VALUE;
     long datasetUpstreamTimestamp = Long.MAX_VALUE;
 
+    final FileSystem sourceFs = getSourceFileSystem(this.state);
```

Comment:
Unfortunately this might not be the source filesystem. DIstcp supports copying data from a third filesystem, so this needs to be read from the path of the origin files. You can look at `gobblin.data.management.copy.writer.FileAwareInputStreamDataWriter` for an example.

Consider having the writer populate this information into the state instead, as the writer is guaranteed to know the source and target filesystems.